File: csv/haskell/update/csv_update.hs
#! /usr/bin/runghc
-- -----------------------------------------------------------------
--	csv_update.hs
--
--						Dec/25/2014
--
-- -----------------------------------------------------------------
import System.Environment
import	System.IO.Strict

import Text_manipulate
-- -----------------------------------------------------------------
main :: IO ()
main = do
	putStrLn "*** 開始 ***"
	args <- getArgs
	let file_name = head args
	let key_in = head (tail args)
	let population_in = head (tail (tail args))
	putStr (key_in ++ "\t")
	putStr (population_in ++ "\n")
--
--
	inStr <- System.IO.Strict.readFile file_name
--
	let dict_aa = csv_to_dict_proc inStr
--
	today <- get_current_date_proc
--
	let dict_bb = dict_update_proc key_in population_in today dict_aa
--
	let str_csv_new = dict_to_str_proc "," dict_bb

	writeFile file_name str_csv_new
--
	putStrLn "*** 終了***"
-- -----------------------------------------------------------------

File: csv/julia/update/csv_update.jl
#! /usr/bin/julia
#
#	csv_update.jl
#
#					Jul/05/2018
# --------------------------------------------------------------------
include("text_manipulate.jl")
#
# --------------------------------------------------------------------
println("*** 開始 ***")
fname_csv = ARGS[1]
key_in = ARGS[2]
population_in = ARGS[3]
println(fname_csv)
println(key_in,'\t',population_in)
#
dict_aa = csv_to_dict_proc(fname_csv)
#
if ( haskey(dict_aa,key_in))
	dict_aa = dict_update_proc(dict_aa,key_in,population_in)
	dict_display_proc(dict_aa)
	dict_to_csv_proc(dict_aa,fname_csv)
end
#
println("*** 終了 ***")
# --------------------------------------------------------------------
